src/components/Planning/PlanningClassesTable/PlanningClassesTableRow.tsx

```diff
@@ -0,0 +1,130 @@
+import { IClass } from '@/interfaces/classInterfaces';
+import { classSums } from '@/mocks/common';
+import { useParams } from 'react-router';
+import { Link } from 'react-router-dom';
+
+export type ClassType = 'masterClass' | 'class' | 'subClass';
+
+interface IPlanningClassesTableRowProps {
+  projectClass: IClass;
+  type: ClassType;
+  children?: ReactNode;
+  initiallyExpanded?: true;
+}
+
+// eslint-disable-next-line react/display-name
+const OverrunSum = memo(({ value }: { value: string }) => (
+  <span>
+    <BubbleIcon value="y" color="white" size="s" />
+    {`+${value}`}
+  </span>
+));
+
+const PlanningClassesTableRow: FC<IPlanningClassesTableRowProps> = ({
+  projectClass,
+  type,
+  children,
+  initiallyExpanded,
+}) => {
+  const { masterClassId, classId } = useParams();
+  const [expanded, setExpanded] = useState(initiallyExpanded || false);
+  const [isHovering, setIsHovering] = useState(false);
+
+  const handleMouseOver = useCallback(() => {
+    setIsHovering(true);
+  }, []);
+
+  const handleMouseOut = useCallback(() => {
+    setIsHovering(false);
+  }, []);
+
+  const handleExpanded = useCallback(() => {
+    setExpanded((current) => !current);
+  }, []);
+
+  useEffect(() => {
+    setExpanded(initiallyExpanded || false);
+  }, [initiallyExpanded]);
+
+  const buildLink = useCallback(() => {
+    switch (type) {
+      case 'masterClass':
+        return projectClass.id;
+      case 'class':
+        return `${masterClassId}/${projectClass.id}`;
+      case 'subClass':
+        return `${masterClassId}/${classId}/${projectClass.id}`;
+    }
+  }, [projectClass.id, masterClassId, classId, type]);
+
+  return (
+    <>
+      <tr>
+        {/* Header with cell name */}
+        <th className={`class-header-cell ${type}`}>
+          <div style={{ position: 'relative' }}>
+            <div className="class-header-content-item value-container">
+              {/* class code/number here */}
+              <span>{}</span>
+            </div>
+            <div className={`class-header-content ${type}`}>
+              <div className="class-header-content-item">
+                <Link to={buildLink()} className="display-flex">
+                  <IconButton
+                    icon={expanded ? IconAngleUp : IconAngleDown}
+                    color="white"
+                    onClick={handleExpanded}
+                  />
+                </Link>
+              </div>
+              <div className="class-header-content-dots">
+                <IconMenuDots size="xs" />
+              </div>
+              <div
+                className="class-header-content-item"
+                onMouseOver={handleMouseOver}
+                onMouseOut={handleMouseOut}
+              >
+                <div className="class-title-container">
+                  <Span fontWeight="bold" size="s" text={projectClass.name} color="white" />
+                </div>
+                {isHovering && (
```

Comment:
We can do this with CSS magic, no need for state variable

Reply:
Resolved